src/main/java/org/terasology/launcher/LauncherInitTask.java

```diff
@@ -192,7 +192,19 @@ private BaseLauncherSettings getLauncherSettings(Path launcherDirectory) throws
         return launcherSettings;
     }
 
-    private boolean checkForLauncherUpdates(Path downloadDirectory, Path tempDirectory, boolean saveDownloadedFiles) {
+    private boolean checkForLauncherUpdates(OperatingSystem os, Path downloadDirectory, Path tempDirectory, boolean saveDownloadedFiles) {
+        final String platform;
+        if (os.isWindows()) {
+            platform = "windows";
+        } else if (os.isMac()) {
+            platform = "mac";
+        } else if (os == OperatingSystem.LINUX) {
+            platform = "linux";
+        } else {
+            logger.warn("Current platform is unsupported: {}", System.getProperty("os.name"));
```

Comment:
`System.getProperty("os.name")` - why not print out the `os` instead? Mabye update the `toString` to contain more information, if needed.

Reply:
That can be a quick fix. :+1: Since we're planning to get rid of that enum or maybe refactor it at least, it should contain just four constants: _windows, mac, linux_, and _unsupported_. So we might need to get it refactored while solving #542 too.

Reply:
Why would we need the `unsupported` constant?

Reply:
That'd be the default if someone somehow tries to run the launcher on anything other than the three supported platforms. We can use `null` of course, but I'd suggest avoiding that.

Reply:
I get what "unsupported" means ;)
I just don't get why we'd need a specific value for that...
Consider you want somebody to slice **apples**, peel **potatoes** and throw **everything else** you find in your kitchen cabinet away, which of the following would you rather write down and why?

A)
```
- open kitchen cabinet and do for each
    - if "apple": slice
    - if "potato": peel
    - if "something else": throw away
```

B)
```
- open kitchen cabinet and do for each
    - if "apple": slice
    - if "potato": peel
    - else: throw away
```

Reply:
(B) seems natural :+1: (and that's what the `else` keyword was invented for)